ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/HighAvailabilityTestUtil.java

```diff
@@ -264,5 +270,121 @@ public static void sendLagReportingRequest(
           .get();
     }
   }
+
+  static String extractQueryId(final String outputString) {
+    final java.util.regex.Matcher matcher = QUERY_ID_PATTERN.matcher(outputString);
+    if (!matcher.find()) {
+      throw new AssertionError("Could not find query id in: " + outputString);
+    }
+    return matcher.group(1);
+  }
+
+  // Ensures that lags exist for the cluster.  Makes the simplified assumption that there's just one
+  // state store.
+  static BiFunction<KsqlHostInfoEntity, Map<KsqlHostInfoEntity, HostStatusEntity>, Boolean>
+  lagsExist(
+      final int expectedClusterSize
+  ) {
+    return (remoteServer, clusterStatus) -> {
+      if (clusterStatus.size() == expectedClusterSize) {
+        int numWithLag = 0;
+        for (Entry<KsqlHostInfoEntity, HostStatusEntity> e : clusterStatus.entrySet()) {
+          if (e.getValue().getHostStoreLags().getStateStoreLags().size() > 0) {
+            numWithLag++;
+          }
+        }
+        if (numWithLag >= Math.min(expectedClusterSize, 2)) {
+          LOG.info("Found expected lags: {}", clusterStatus.toString());
+          return true;
+        }
+      }
+      LOG.info("Didn't yet find expected lags: {}", clusterStatus.toString());
+      return false;
+    };
+  }
+
+  // Ensures that lags exist for the given host.  Makes the simplified assumption that there's just
+  // one state store.
+  static BiFunction<KsqlHostInfoEntity, Map<KsqlHostInfoEntity, HostStatusEntity>, Boolean>
+  lagsExist(
+      final int clusterSize,
+      final KsqlHostInfoEntity server,
+      final long currentOffset,
+      final long endOffset
+  ) {
+    return (remote, clusterStatus) -> {
+      if (clusterStatus.size() == clusterSize) {
+        HostStatusEntity hostStatusEntity = clusterStatus.get(server);
+        if (hostStatusEntity == null) {
+          LOG.info("Didn't find {}", server.toString());
+          return false;
+        }
+        Pair<Long, Long> pair = getOffsets(server,clusterStatus);
+        long current = pair.left;
+        long end = pair.right;
+        if ((currentOffset < 0 || current >= currentOffset) && end >= endOffset) {
```

Comment:
This method checks for a specific server if there is actual lag reported? I don't understand the if condition: `currentOffset < 0 ` what does this mean?

Reply:
If offset currentOffset < 0, we don't check it.  It's just mean as a special value since some tests don't care about this.  I'll make this optional instead to make it more obvious.